contracts/exchangeIssuance/ExchangeIssuanceZeroEx.sol

```diff
@@ -0,0 +1,561 @@
+        emit ExchangeIssue(msg.sender, _setToken, _inputToken, _maxAmountInputToken, _amountSetToken);
+        return totalInputTokenSold;
+    }
+
+
+    /**
+    * Issues an exact amount of SetTokens for given amount of ETH.
+    * The excess amount of tokens is returned in an equivalent amount of ether.
+    *
+    * @param _setToken              Address of the SetToken to be issued
+    * @param _amountSetToken        Amount of SetTokens to issue
+    * @param _componentQuotes       The encoded 0x transactions to execute
+    *
+    * @return amountEthReturn       Amount of ether returned to the caller
+    */
+    function issueExactSetFromETH(
+        ISetToken _setToken,
+        uint256 _amountSetToken,
+        bytes[] memory _componentQuotes,
+        address _issuanceModule
+    )
+        isWhitelistedIssuanceModule(_issuanceModule)
+        external
+        nonReentrant
+        payable
+        returns (uint256)
+    {
+        require(msg.value > 0, "ExchangeIssuance: INVALID ETH AMOUNT");
+
+        IWETH(WETH).deposit{value: msg.value}();
+        _safeApprove(IERC20(WETH), swapTarget, msg.value);
+
+        uint256 totalEthSold = _buyComponentsForInputToken(_setToken, _amountSetToken, _componentQuotes, IERC20(WETH), _issuanceModule);
+
+        require(totalEthSold<= msg.value, "ExchangeIssuance: OVERSPENT ETH");
+        IBasicIssuanceModule(_issuanceModule).issue(_setToken, _amountSetToken, msg.sender);
+
+        uint256 amountEthReturn = msg.value.sub(totalEthSold);
+        if (amountEthReturn > 0) {
+            IWETH(WETH).withdraw(amountEthReturn);
+            payable(msg.sender).sendValue(amountEthReturn);
+        }
+
+        emit ExchangeIssue(msg.sender, _setToken, IERC20(ETH_ADDRESS), totalEthSold, _amountSetToken);
+        return amountEthReturn; 
+    }
+
+    /**
+     * Redeems an exact amount of SetTokens for an ERC20 token.
+     * The SetToken must be approved by the sender to this contract.
+     *
+     * @param _setToken             Address of the SetToken being redeemed
+     * @param _outputToken          Address of output token
+     * @param _amountSetToken       Amount SetTokens to redeem
+     * @param _minOutputReceive     Minimum amount of output token to receive
+     * @param _componentQuotes      The encoded 0x transactions execute (components -> WETH).
+     *
+     * @return outputAmount         Amount of output tokens sent to the caller
+     */
+    function redeemExactSetForToken(
+        ISetToken _setToken,
+        IERC20 _outputToken,
+        uint256 _amountSetToken,
+        uint256 _minOutputReceive,
+        bytes[] memory _componentQuotes,
+        address _issuanceModule
+    )
+        isWhitelistedIssuanceModule(_issuanceModule)
+        external
+        nonReentrant
+        returns (uint256)
+    {
+
+        uint256 outputAmount;
+        _redeemExactSet(_setToken, _amountSetToken, _issuanceModule);
+
+        // Liquidate components for WETH and ignore _outputQuote
```

Comment:
What does "ignore _outputQuote" refer to here? Is this still true?

Reply:
This is outdated and refered to a  previous implementation when we swapped the components to WETH and then WETH to input/output token in separate 0x trades.

Reply:
Removed it